tests/lib/snaps/test-snapd-service-watchdog/bin/direct

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python3
+
+import socket
+import time
+import os
+import argparse
+import logging
+
+from contextlib import closing
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(description='watchdog notification test')
+    parser.add_argument('--bad', action='store_true', help='bad mode')
+    return parser.parse_args()
+
+
+def main(opts):
+    notify_addr = os.getenv('NOTIFY_SOCKET')
+    if not notify_addr:
+        raise RuntimeError('NOTIFY_SOCKET not set')
+
+    watchdog_usec = os.getenv('WATCHDOG_USEC')
+    if not watchdog_usec:
+        watchdog_timeout = 10
+        logging.warning('WATCHDOG_USEC not set, assuming default value')
```

Comment:
There's a known protocol established by systemd that once the watchdog is in place, that environment variable should be set to tell the daemon it needs to ping or otherwise be killed. So we need to ensure that the daemon indeed gets this error, or fail the test, instead of going on with a default value.